vivisect/impemu/emulator.py

```diff
@@ -230,36 +232,49 @@ def checkBranches(self, starteip, endeip, op):
         entries to the current path, and updates current path as needed
         (returns a list of (va, CodePath) tuples.
         """
-
-        paths = set()
+        '''
+        So I've kinda gone back and forth on this one a bit. On one hand, the emulator
+        should be free to iterate over things as it needs. On the other, it's operating with
+        imperfect information as to what's a valid path and what's not.
```

Comment:
Yea, it's a weird contention of "we want to be able to discover unexpected things" so that we're permissive about where the emulator can run off to. The problem is that sometimes we're too permissive, especially if we haven't bounded consecutive jump tables, we end up correlating them together and that ends up causing all kinds of trouble.

So introduce a new API that returns a super stripped down, lightweight emulator that knows how to execute opcodes and that's about it and just runs off that, and have the WorkspaceEmulator use all the analysis results at it's disposal? I like it.

Reply:
yeah...  about that. (unbounded consecutive jump tables)
you and i should sit down in front of a fire, make some hot cocoa, and read through the symbolik_switchcase PR together.  read it together like a book.  enjoy the essence of it.  talk about what emotions it brings up.  bond over it.

ok, so however weird that just made it, we should do it.  it has a lot of ideas that would make life better.  it's far from perfect, and can certainly use improvements and tweaks.  but i feel like it's a lot to chew on alone... and would be better looking through it together in person or via vidconf

i'm not distracting you!  we're on a path:
* pypi
* PowerPC
* Py3 / Symbolik_Switchcase

Reply:
Yea, I've just been holding off on it since ARM was like 99% of the way there until just recently and pypi just became a priority.

But when all that blows over I'll actually grab some coffee, and nice big comfy chair, and properly read through it. I'm more trying to keep things relatively sane until we can get that in there.

Speaking of pypi, I've this and your ELFPLT one marked down as the last two things needed to get that out the door and a initial review on your branch. So we're close :D